test/src/test/java/org/corfudb/integration/CorfuReplicationReconfigurationIT.java

```diff
@@ -215,6 +227,172 @@ public void testActiveClusterReset() throws Exception {
+                            null,
+                            TableOptions.fromProtoSchema(LogReplicationMetadata.ReplicationEvent.class));
+
+            Table<StringKey, IntValue, Metadata>  mapActive = corfuStoreActive.openTable(
+                    NAMESPACE,
+                    testStreamName,
+                    StringKey.class,
+                    IntValue.class,
+                    Metadata.class,
+                    TableOptions.builder().schemaOptions(
+                                    CorfuOptions.SchemaOptions.newBuilder()
+                                            .setIsFederated(true)
+                                            .addStreamTag(ObjectsView.LOG_REPLICATOR_STREAM_INFO.getTagName())
+                                            .build())
+                            .build()
+            );
+
+            Table<StringKey, IntValue, Metadata>  mapStandby = corfuStoreStandby.openTable(
+                    NAMESPACE,
+                    testStreamName,
+                    StringKey.class,
+                    IntValue.class,
+                    Metadata.class,
+                    TableOptions.builder().schemaOptions(
+                                    CorfuOptions.SchemaOptions.newBuilder()
+                                            .setIsFederated(true)
+                                            .addStreamTag(ObjectsView.LOG_REPLICATOR_STREAM_INFO.getTagName())
+                                            .build())
+                            .build()
+            );
+
+            // Write batchSize num of entries to active map
+            for (int i = 0; i < batchSize; i++) {
+                try (TxnContext txn = corfuStoreActive.txn(NAMESPACE)) {
+                    txn.putRecord(mapActive, StringKey.newBuilder().setKey(String.valueOf(i)).build(),
+                            IntValue.newBuilder().setValue(i).build(), null);
+                    txn.commit();
+                }
+            }
+            assertThat(mapActive.count()).isEqualTo(batchSize);
+            assertThat(mapStandby.count()).isZero();
+
+            startActiveLogReplicator();
+            startStandbyLogReplicator();
+            log.info("Replication servers started, and replication is in progress...");
+
+            // Wait for replication to finish and check sink has all the writes
+            boolean entriesReplicated = false;
+            for(int i = 0; i < PARAMETERS.NUM_ITERATIONS_MODERATE; i++) {
+                if (mapStandby.count() == mapActive.count()) {
+                    entriesReplicated = true;
+                    break;
+                }
+                Thread.sleep(50);
+            }
+            assertThat(entriesReplicated).isTrue();
+
+            // Check that the event table is empty to begin
+            assertThat(eventTable.count()).isZero();
+
+            // Add some empty event to act as older stale events.
```

Comment:
Wouldnt these events cause a snapshot sync before the new event is written to the config table?

Reply:
Yes that would be the case, can update the comments to reflect. Functionally should be the same though because it is unlikely any of those events finish before the one in the config table is enqueued.

Reply:
how can we be sure that the event through the config table was processed first?  In fact, these events will be processed in sequence.  Let me know if I am missing something.

Reply:
when we enqueue a new event we receive a sync cancel for the current, so if they are triggered back-to-back like in this case only the last one should run until completion right?

Reply:
Ok I see.

However, event listener and the replication fsm run in separate threads so there is no guarantee that the previous events have not already completed by the time the last one is processed and snapshot sync request is made.

One way to add the guarantee is to add an empty event type(empty message) to the table initially.  These types will be filtered out, the last sync executed and it will clear the table.

Reply:
Updated. Saw that adding the event with an empty cluster id has the same effect due to [this](https://github.com/CorfuDB/CorfuDB/blob/7cb27ec468f4a0178be01d863d94351a35a4c76d/infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/CorfuReplicationManager.java#L251) check